### PR TITLE
Add input and accesslist to mock tx

### LINKED ISSUE #4901: Add AccessList and input fields to MockTransaction
### Describe the feature

The MockTransaction variants currently don't have input/tx list fields

https://github.com/paradigmxyz/reth/blob/081d71e1a2f6ae3731c9207a7533beb98861ca8c/crates/transaction-pool/src/test_utils/mock.rs#L84-L116

## TODO
* add accesslist field
* add input fields

this should mirror the actual transaction enum variants

https://github.com/paradigmxyz/reth/blob/081d71e1a2f6ae3731c9207a7533beb98861ca8c/crates/primitives/src/transaction/mod.rs#L59-L59

do you want to take this @DoTheBestToGetTheBest ?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
Hello everyone i hope all are goods, i'am looking for hints on how i can continue this issue

first i'm asking myself if it was the right thing to make accesslist element as vector for AccessList struct or could be better if i directly make it AccessList ? And any other suggestion and help could be really goods

Thank you so much !

### CODE REVIEW

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 1]
```diff
@@ -101,6 +102,8 @@ pub enum MockTransaction {
         gas_limit: u64,
         to: TransactionKind,
         value: U256,
+        accesslist: Vec<AccessList>,
+        input: Vec<u8>,
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:106]: we want 
```suggestion
        accesslist: AccessList,
        input: Bytes,
```

here

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 2]
```diff
@@ -541,6 +551,8 @@ impl FromRecoveredTransaction for MockTransaction {
                 gas_limit,
                 to,
                 value: value.into(),
+                accesslist: todo!(),
+                input: todo!(),
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:555]: we need to use the values from Line 543-544

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/transaction-pool/src/test_utils/mock.rs (hunk 2 above)
Comment (DoTheBestToGetTheBest) [reply crates/transaction-pool/src/test_utils/mock.rs:555]: > we need to use the values from Line 543-544

Hello again @mattsse i'm really sorry i was in school. I'm back now. I still don't get the idea on how to implement this part. Could you please if you have time tell me more about it ? Im sorry if my questions can be stupid

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse hey again, i think i done. Could you please check and tell me if it's good please?

**Review by mattsse — APPROVED** [review]
ty

**Pull request merged**